spicy/toolchain/src/compiler/resolver.cc

```diff
@@ -128,6 +128,35 @@ struct VisitorPass2 : visitor::MutatingPostOrder {
             else
                 n->addError(x.error());
         }
+
+        else if ( n->tag() == "&requires" ) {
+            if ( ! n->hasValue() )
+                // Caught elsewhere, we don't want to report it here again.
+                return;
+
+            auto cond = *n->valueAsExpression();
+            if ( ! cond->isResolved() )
+                return;
+
+            if ( cond->type()->type()->isA<hilti::type::Result>() )
+                return;
+
+            auto ne = coerceExpression(builder(), cond,
+                                       builder()->qualifiedType(builder()->typeBool(), hilti::Constness::Const));
+            if ( ! ne.coerced ) {
+                n->addError(ne.coerced.error());
+                return;
+            }
+
+            // Implicitly create an error message from the condition itself.
+            std::stringstream ss;
+            ss << *cond;
+            auto msg = hilti::util::fmt("&requires failed: %s", hilti::util::replace(ss.str(), "__dd", "$$"));
```

Comment:
We can directly stringify the node and avoid the `stringstream`,

```suggestion
             auto msg = hilti::util::fmt("&requires failed: %s", hilti::util::replace(cond->print(), "__dd", "$$"));
```